src/index.ts

```diff
@@ -126,7 +125,11 @@ function initializeProvider(providerOrRpcUrl, gasInfo) {
 
   // if provider is a string, assume it's a rpcUrl
   if (typeof providerOrRpcUrl === 'string') {
-    return new ProviderWrapper(new HttpProvider(providerOrRpcUrl), gasInfo);
+    const client = createPublicClient({
+      chain: lukso,
```

Comment:
hmmmm but if I want to use that on LUKSO Testnet, then it would not work because you are passing mainnet as `chain`?

Reply:
The configured public client could provide the chain, but this is just the wrapper to allow specification of the URL. It will connect to testnet just fine, the chain is mainly used to get the URL (which is specificed) the multicall contracts (which is correct but we're not using anyways) and the explorer which we're not using in this scenario.